lib/vocabularies/discriminator/index.ts

```diff
@@ -57,27 +65,72 @@ const def: CodeKeywordDefinition = {
       return _valid
     }
 
+    function isRef(schema: object) {
+      return Object.keys(schema).includes("$ref")
+    }
+
+    function findDiscriminatorTagName(
+      schema: AnySchemaObject,
+      tagName: string
+    ): AnySchemaObject | undefined {
+      const sch = isRef(schema)
+        ? loadSchemaSync?.(it.baseId, schema["$ref"], schema.id || schema.$id || "")
+        : schema
+      if (!sch) {
+        throw new Error(`Cannot resolve reference ${schema["$ref"]}`)
+      }
+      const schemaObject = sch as AnySchemaObject
+      if (schemaObject.properties && schemaObject.properties[tagName]) {
+        return {
+          id: schemaObject["$id"],
+          schema: schemaObject.properties[tagName],
+          required: schemaObject.required,
```

Comment:
I would simplify return here:

```suggestion
          schema: schemaObject,
          propSchema: schemaObject.properties[tagName]
```

Then you can use it below like:

const { propSchema, schema } = findDiscriminatorTagName(sch, tagName);

and it will be a bit more readable. Because right now it's hard to follow what is what.